kaldb/src/main/java/com/slack/kaldb/server/OpenSearchBulkIngestApi.java

```diff
@@ -97,6 +99,9 @@ public void load() {
       List<DatasetMetadata> datasetMetadataToProcesses =
           filterValidDatasetMetadata(datasetMetadataList);
 
+      checkArgument(
```

Comment:
These should be `checkState` instead of `checkArgument` - the argument version is only intended for checking params to the calling method.

```
  /**
   * Ensures the truth of an expression involving one or more parameters to the calling method.
   *
   * @param expression a boolean expression
   * @throws IllegalArgumentException if {@code expression} is false
   */
  public static void checkArgument(boolean expression) {
```